search/scorer/scorer_term.go

```diff
@@ -61,19 +62,36 @@ func (s *TermQueryScorer) Size() int {
 	return sizeInBytes
 }
 
-func NewTermQueryScorer(queryTerm []byte, queryField string, queryBoost float64, docTotal, docTerm uint64, options search.SearcherOptions) *TermQueryScorer {
+func (s *TermQueryScorer) computeIDF(avgDocLength float64, docTotal, docTerm uint64) float64 {
+	var rv float64
+	if avgDocLength > 0 {
+		// avgDocLength is set only for bm25 scoring
+		rv = math.Log(1 + (float64(docTotal)-float64(docTerm)+0.5)/
+			(float64(docTerm)+0.5))
+	} else {
+		rv = 1.0 + math.Log(float64(docTotal)/
+			float64(docTerm+1.0))
+	}
+
+	return rv
+}
+
+func NewTermQueryScorer(queryTerm []byte, queryField string, queryBoost float64, docTotal,
+	docTerm uint64, avgDocLength float64, options search.SearcherOptions) *TermQueryScorer {
```

Comment:
I realize `docTerm` is carry-over name, but perhaps we should replace all occurrences of it with something better indicative of what it exactly is.

Reply:
maybe the name is self-explanatory (the alternate names i can think of is docFreq - number of documents having the term). it's just that the lack commentary makes it a little less obvious. shall i just add the commentary alongside docTerm in the struct and leave it as is?

Reply:
Roger, perhaps some commentary on all the function parameters then.